### PR TITLE
prov/efa: fix bug in handling FI_SELECTIVE_COMPLETION

### PR DESCRIPTION
**Pull request by wzamazon**
This PR contain 2 commits:

1. use tx_attr->op_flags as the default flag when calling rxr_msg_generic_send()
2. fix the logic of writing completion entry in rxr_cq_write_tx_completion() and rxr_cq_write_rx_completion()

### CODE REVIEW

File: prov/efa/src/rxr/rxr_pkt_type_misc.c [hunk 1]
```diff
@@ -385,7 +385,12 @@ void rxr_pkt_handle_rma_read_completion(struct rxr_ep *ep,
 		if (read_entry->x_entry_type == RXR_TX_ENTRY) {
 			tx_entry = ofi_bufpool_get_ibuf(ep->tx_entry_pool, read_entry->x_entry_id);
 			assert(tx_entry && tx_entry->cq_entry.flags & FI_READ);
-			rxr_cq_write_tx_completion(ep, tx_entry);
+			if (tx_entry->fi_flags & FI_COMPLETION) {
```
Comment (rajachan) [inline prov/efa/src/rxr/rxr_pkt_type_misc.c:388]: We seem to be doing this check anytime we call `rxr_cq_write_tx_completion`, can we move the selective completion logic inside it? We have the counter logic in there too.

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_type_misc.c:388]: Good idea. will do

File: prov/efa/src/rxr/rxr_cq.c [hunk 2]
```diff
@@ -724,14 +724,34 @@ void rxr_cq_handle_shm_completion(struct rxr_ep *ep, struct fi_cq_data_entry *cq
...
+
+
 void rxr_cq_write_tx_completion(struct rxr_ep *ep,
 				struct rxr_tx_entry *tx_entry)
 {
 	struct util_cq *tx_cq = ep->util_ep.tx_cq;
 	int ret;
 
-	if (!(tx_entry->fi_flags & RXR_NO_COMPLETION) &&
-	    ofi_need_completion(rxr_tx_flags(ep), tx_entry->fi_flags)) {
```
Comment (rajachan) [inline prov/efa/src/rxr/rxr_cq.c:734]: Hmm, don't we need the ofi_need_completion checks too?

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_cq.c:734]: ofi_need_completion check  ```(cq_flags & FI_SELECTIVE_COMPLETION)``` and ```(op_flags | FI_COMPLETION)```.

the information of FI_SELECTIVE_COMPLETION is contained in ```util_ep.tx_msg_attr```, the information of ```op_flags``` is contained in ```tx_entry->flags```, so ```rxr_cq_need_tx_completion``` covers all aspects of ```ofi_need_completion```.

**Comment by wzamazon** [comment]
bot:aws:retest

File: prov/efa/src/rxr/rxr_cq.c (hunk 2 above)
Comment (rajachan) [reply prov/efa/src/rxr/rxr_cq.c:734]: Talked to Wei offline. I was missing the fact that we were passing tx_op_flags as cq_flag, erroneously.

**Comment by rajachan** [comment]
This would need to be backported to 1.10.x too.

**Comment by rajachan** [comment]
The NCCL job failure in AWS-CI is due to a test environment issue that's being fixed.

**Pull request merged**